src/cljs/kixi/nhs/exploration_ui/dataset_details.cljs

```diff
@@ -55,13 +70,73 @@
        [:tbody
         (om/build-all resource-item cursor {:key :id})]]]])))
 
-(defn dataset-preview
-  ""
+(defmulti validate-selection (fn [owner header axis] axis))
+
+(defmethod validate-selection :none [owner header _]
+  (when (= header (:field-name (om/get-state owner :x_axis)))
+    (om/set-state! owner :x_axis {})
+    (om/set-state! owner :y_axis (vec (remove #(= (:field %) header) (om/get-state owner :y_axis))))))
+
+(defmethod validate-selection :x_axis [owner header _]
+  (let [current-selection (om/get-state owner :x_axis)]
+    (if (seq current-selection)
+      (om/set-state! owner :error {:message "You can only select a single field to plot on X axis."})
+      (om/set-state! owner :x_axis {:field-name header}))))
+
+(defmethod validate-selection :y_axis [owner header _]
+  (om/set-state! owner :y_axis (conj (om/get-state owner :y_axis) {:field-name header})))
+
+(defn on-change [owner header e]
+  (let [v (.-value (aget (.-options (.-target e)) (.-selectedIndex (.-options (.-target e)))))]
```

Comment:
pull out options as a separate thing as it is being retrieved twice and I've spent a couple of minute puzzling over what this is doing (though being a salesman may have damaged my brain today)